delta-api/src/main/java/io/cdap/delta/api/DMLEvent.java

```diff
@@ -182,8 +190,56 @@ public Builder setRowId(String rowId) {
     }
 
     public DMLEvent build() {
-      return new DMLEvent(offset, database, new DMLOperation(table, operationType, ingestTimestampMillis), row,
-                          previousRow, transactionId, ingestTimestampMillis, sourceTimestampMillis, isSnapshot, rowId);
+      int sizeInBytes = (operationType == DMLOperation.Type.INSERT ||
+        operationType == DMLOperation.Type.UPDATE) ? computeSizeInBytes(row) : 0;
+
+      return new DMLEvent(offset, database, new DMLOperation(table, operationType, ingestTimestampMillis, sizeInBytes),
+                          row, previousRow, transactionId, ingestTimestampMillis, sourceTimestampMillis, isSnapshot,
+                          rowId);
+    }
+
+    private int computeSizeInBytes(StructuredRecord row) {
+      List<Schema.Field> fields = row.getSchema().getFields();
+      if (fields == null) {
+        return 0;
+      }
+      int sizeInBytes = 0;
+      for (Schema.Field field : fields) {
+        Schema fieldSchema = field.getSchema();
+        fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+        if (row.get(field.getName()) == null) {
+          continue;
+        }
+        switch (fieldSchema.getType()) {
+          case BOOLEAN:
+            sizeInBytes += 1;
+            break;
+          case INT:
+          case FLOAT:
+          case ENUM:
+            sizeInBytes += 4;
+            break;
+          case LONG:
+          case DOUBLE:
+            sizeInBytes += 8;
+            break;
+          case BYTES:
+            Object value = row.get(field.getName());
```

Comment:
nit: can have this variable before the null check (line 210) to do just a single get instead of 2.